CaT/src/TableRelations/Tables/DerivedField.php

```diff
@@ -32,6 +32,19 @@ public function derivedFrom() {
 	}
 
 	/**
+	 * Get first order of fields from which this field is derived.
+	 *
+	 * @return	AbstractTableField[]
+	 */
+	public function derivedFromDirect() {
+		$return = array();
+		foreach ($this->derived_from as $field) {
+			$return[] = $field;
+		}
```

Comment:
Why this? I think you could just `return $this->derived_from`. Which also makes the name `derived_from` questionable, regarding the stuff that is done in the method with the similar name.